File: remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int n = nums.length;
        if (n == 1) return 1;
        int curr = nums[0];
        boolean seen = false;
        Queue<Integer> uniques = new LinkedList<>();
        //ArrayList<Integer> uniques = new ArrayList<>();
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] != curr) {
                uniques.add(nums[i]);
                curr = nums[i];
            }
        }
        int i = 1;
        while (!uniques.isEmpty()) {
            nums[i++] = uniques.poll();
        }
        return i;
    }
}

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        p = 1
        for i in range(1, len(nums)):
            if nums[i] > nums[i - 1]:
                nums[p] = nums[i]
                p += 1
        return p